spec/moonshot/plugins/backup_spec.rb

```diff
@@ -32,6 +32,17 @@
       expect { subject.new }.to raise_error(ArgumentError)
     end
 
+    it 'should raise ArgumentError if redundant parameters are provided' do
+      expect do
+        subject.new do |b|
+          b.bucket = 'test'
+          b.buckets = {}
+          b.files = %w(sample files)
+          b.hooks = [:sample, :hooks]
+        end
+      end.to raise_error(ArgumentError)
+    end
+
     let(:backup) do
```

Comment:
Do we want to add better test coverage of the bucket behavior in the hash case, e.g. if no account alias is set or no account alias matches it would be nice to assert this behavior.